openedx_events/event_bus/avro/deserializer.py

```diff
@@ -51,6 +51,17 @@ def _deserialized_avro_record_dict_to_object(data: dict, data_type, deserializer
         # check whether list items type is in basic types.
         if arg_data_type[0] in SIMPLE_PYTHON_TYPE_TO_AVRO_MAPPING:
             return data
+    elif data_type_origin == dict:
+        # returns types of dict contents
+        # if data_type == Dict[str, int], arg_data_type = (str, int)
+        arg_data_type = get_args(data_type)
+        if not arg_data_type:
+            raise TypeError(
+                "Dict without annotation type is not supported. The argument should be a type, for eg., Dict[str, int]"
+            )
+        # check whether dict items type is in basic types.
+        if arg_data_type[1] in SIMPLE_PYTHON_TYPE_TO_AVRO_MAPPING:
+            return data
```

Comment:
Do we also need to check whether the dict key type is a simple Python type?

Reply:
I added the check to study the behavior, and I got a few errors because the keys are unhashable by nature. Therefore, only basic Python types were allowed. Do you think we need an explicit check for this?

Reply:
Can you say more about keys being unhashable? What's a situation where that happens?

Reply:
@timmc-edx: I implemented your suggestion in this commit https://github.com/openedx/openedx-events/pull/433/commits/7eabdaa2304e4b143a86e88db7619c20f50a7f22